src/frontends/sql/decoder_context.cc

```diff
@@ -9,61 +13,51 @@ using ir::NamedValueMap;
 using ir::Operation;
 using ir::Value;
 
-static NamedValueMap MakeNamedValueMapFromVector(absl::string_view prefix,
-                                                 std::vector<Value> vec,
-                                                 NamedValueMap map) {
-  uint64_t i = 0;
-  for (auto iter = std::make_move_iterator(vec.begin());
-       iter != std::make_move_iterator(vec.end()); ++iter) {
-    auto insert_result = map.insert({absl::StrCat(prefix, i), *iter});
-    CHECK(insert_result.second)
-        << "Collision on key " << insert_result.first->first;
-    ++i;
-  }
-  return map;
-}
-
 const Operation &DecoderContext::MakeMergeOperation(
     std::vector<Value> direct_inputs, std::vector<Value> control_inputs) {
-  // Add the direct and control inputs into the merge with predictable
-  // prefixes.
+  size_t control_start_index = direct_inputs.size();
+  CHECK(control_start_index < std::numeric_limits<int64_t>::max());
+  auto attributes =
+      ir::NamedAttributeMap({{std::string(kMergeOpControlStartIndex),
+                              ir::Attribute::Create<ir::Int64Attribute>(
+                                  static_cast<int64_t>(control_start_index))}});
```

Comment:
Optionally, it might be helpful to move this logic into a helper factory on `Int64Attribute`, called something like `CreateFromUint64` or something. That way you don't need to worry about it at every callsite.

Reply:
SG. I will send this out as separate PR so that I can add some tests.

Reply:
Good Idea, btw.